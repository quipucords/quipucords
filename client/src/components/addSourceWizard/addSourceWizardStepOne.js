import React from 'react';
import PropTypes from 'prop-types';
import { Form, Radio } from 'patternfly-react';
import { connect, store, reduxSelectors, reduxTypes } from '../../redux';
import { FormState } from '../formState/formState';
import apiTypes from '../../constants/apiConstants';

class AddSourceWizardStepOne extends React.Component {
  isStepValid = ({ values }) => {
    store.dispatch({
      type: reduxTypes.sources.VALID_SOURCE_WIZARD_STEPONE,
      source: {
        [apiTypes.API_SUBMIT_SOURCE_SOURCE_TYPE]: values.sourceType
      }
    });
  };

  render() {
    const { type } = this.props;

    return (
      <FormState validateOnMount setValues={{ sourceType: type }} validate={this.isStepValid}>
        {({ values, handleOnEvent, handleOnSubmit }) => (
          <Form horizontal onSubmit={handleOnSubmit}>
            <h3 className="right-aligned_basic-form">Select source type</h3>
            <Form.FormGroup>
              <Radio
                name="sourceType"
                value="network"
                checked={values.sourceType === 'network'}
                onChange={handleOnEvent}
              >
                Network Range
              </Radio>
              <Radio
                name="sourceType"
                value="satellite"
                checked={values.sourceType === 'satellite'}
                onChange={handleOnEvent}
              >
                Satellite
              </Radio>
              <Radio
                name="sourceType"
                value="vcenter"
                checked={values.sourceType === 'vcenter'}
                onChange={handleOnEvent}
              >
                vCenter Server
              </Radio>
            </Form.FormGroup>
          </Form>
        )}
      </FormState>
    );
  }
}

AddSourceWizardStepOne.propTypes = {
  type: PropTypes.string
};

AddSourceWizardStepOne.defaultProps = {
  type: 'network'
};

const makeMapStateToProps = () => {
  const mapSource = reduxSelectors.sources.makeSourceDetail();

  return (state, props) => ({
    ...mapSource(state, props)
  });
};

const ConnectedAddSourceWizardStepOne = connect(makeMapStateToProps)(AddSourceWizardStepOne);

export { ConnectedAddSourceWizardStepOne as default, ConnectedAddSourceWizardStepOne, AddSourceWizardStepOne };
