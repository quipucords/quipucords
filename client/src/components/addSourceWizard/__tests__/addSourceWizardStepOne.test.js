import React from 'react';
import configureMockStore from 'redux-mock-store';
import { mount, shallow } from 'enzyme';
import { ConnectedAddSourceWizardStepOne, AddSourceWizardStepOne } from '../addSourceWizardStepOne';

describe('AddSourceWizardStepOne Component', () => {
  const generateEmptyStore = (obj = {}) => configureMockStore()(obj);

  it('should render a connected component', () => {
    const store = generateEmptyStore({ addSourceWizard: {} });
    const component = shallow(<ConnectedAddSourceWizardStepOne />, { context: { store } });

    expect(component.dive()).toMatchSnapshot('connected');
  });

  it('should render a non-connected component', () => {
    const props = {};

    const component = mount(<AddSourceWizardStepOne {...props} />);

    expect(component.render()).toMatchSnapshot('non-connected');
  });
});
