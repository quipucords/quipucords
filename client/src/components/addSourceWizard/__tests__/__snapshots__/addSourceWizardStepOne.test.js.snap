// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connected 1`] = `
<FormState
  onReset={[Function]}
  onSubmit={[Function]}
  resetUsingSetValues={true}
  setValues={
    Object {
      "sourceType": "network",
    }
  }
  setValuesAssumeBoolIsChecked={true}
  setValuesOnUpdate={false}
  validate={[Function]}
  validateOnMount={true}
  validateOnUpdate={false}
>
  <Component />
</FormState>
`;

exports[`non-connected 1`] = `
<form
  class="form-horizontal"
>
  <h3
    class="right-aligned_basic-form"
  >
    Select source type
  </h3>
  <div
    class="form-group"
  >
    <div
      class="radio"
    >
      <label
        title=""
      >
        <input
          checked=""
          name="sourceType"
          type="radio"
          value="network"
        />
        Network Range
      </label>
    </div>
    <div
      class="radio"
    >
      <label
        title=""
      >
        <input
          name="sourceType"
          type="radio"
          value="satellite"
        />
        Satellite
      </label>
    </div>
    <div
      class="radio"
    >
      <label
        title=""
      >
        <input
          name="sourceType"
          type="radio"
          value="vcenter"
        />
        vCenter Server
      </label>
    </div>
  </div>
</form>
`;
